### PR TITLE
Skewed similarity matrix and OpenMP Parallel LocalAligner

### PR DESCRIPTION
**Pull request by huanglangwen**

### CODE REVIEW

**Comment by spaceben** [comment]
Just outlining specifically that this point that this breaks the possibility of  non-linear gap penalties.

**Comment by spaceben** [comment]
Would appreciate some very basic documentation of how exactly the transformation is done, so I can write it down in plain English (or maybe post the picture if you re following that logic still). Thanks man.

**Comment by huanglangwen** [comment]
> 
> 
> Just outlining specifically that this point that this breaks the possibility of non-linear gap penalties.

I think it would be worth to implement another class for nonlinear gap for performance reason.

**Comment by huanglangwen** [comment]
> 
> 
> Would appreciate some very basic documentation of how exactly the transformation is done, so I can write it down in plain English (or maybe post the picture if you re following that logic still). Thanks man.

Well, it might be easier to explain to you tomorrow with some graphs.

**Comment by huanglangwen** [comment]
> 
> 
> > Just outlining specifically that this point that this breaks the possibility of non-linear gap penalties.
> 
> I think it would be worth to implement another class for nonlinear gap for performance reason.

It is still possible though to implement iteration for nonlinear gap functions by branching in similarity matrix class.

**Comment by huanglangwen** [comment]
Here is a demo: (run test on verbose mode)
```
Before Transform:
 0  1  2  3  4  5  6  7
 1  2  3  4  5  6  7  8
 2  3  4  5  6  7  8  9
 3  4  5  6  7  8  9 10
 4  5  6  7  8  9 10 11
 5  6  7  8  9 10 11 12
 6  7  8  9 10 11 12 13
 7  8  9 10 11 12 13 14
 8  9 10 11 12 13 14 15
 9 10 11 12 13 14 15 16
After Transform:
 0  1  2  3  4  5  6  7  8  9
10  1  2  3  4  5  6  7  8  9
10 11  2  3  4  5  6  7  8  9
10 11 12  3  4  5  6  7  8  9
10 11 12 13  4  5  6  7  8  9
10 11 12 13 14  5  6  7  8  9
10 11 12 13 14 15  6  7  8  9
10 11 12 13 14 15 16  7  8  9
```

**Comment by huanglangwen** [comment]
100% acceleration on 8 cores for sw_solve_small.

**Comment by kosta777** [comment]
The code looks good.
